modules/explorer-api/src/main/scala/org/ergoplatform/explorer/http/api/v1/routes/ErgoTreeRoutes.scala

```diff
@@ -0,0 +1,40 @@
+package org.ergoplatform.explorer.http.api.v1.routes
+
+import cats.MonadError
+import cats.effect.{Concurrent, ContextShift, Timer}
+import cats.syntax.semigroupk._
+import io.chrisdavenport.log4cats.Logger
+import org.ergoplatform.explorer.HexString
+import org.ergoplatform.explorer.http.api.ApiErr
+import org.ergoplatform.explorer.http.api.algebra.AdaptThrowable.AdaptThrowableEitherT
+import org.ergoplatform.explorer.http.api.syntax.adaptThrowable._
+import org.ergoplatform.explorer.http.api.syntax.routes._
+import org.ergoplatform.explorer.http.api.v1.defs.ErgoTreeEndpointDefs
+import org.ergoplatform.explorer.http.api.v1.models.ErgoTreeHuman
+import org.ergoplatform.explorer.http.api.models.PrettyErgoTree
+import org.http4s.HttpRoutes
+import sttp.tapir.server.http4s._
+
+
+final class ErgoTreeRoutes[
+  F[_]: Concurrent: ContextShift: Timer: AdaptThrowableEitherT[*[_], ApiErr]
+]()(implicit opts: Http4sServerOptions[F, F], F: MonadError[F, Throwable]) {
+
+  val defs = new ErgoTreeEndpointDefs
+
+  val routes: HttpRoutes[F] = convertErgoTreeR
+
+  private def interpreter = Http4sServerInterpreter(opts)
+
+  private def convertErgoTreeR: HttpRoutes[F] =
+    interpreter.toRoutes(defs.convertErgoTreeDef) { req => {
+      val hexErgoTree = HexString.fromStringUnsafe(req.hashed)
+      val (ergoTreeConstants, ergoTreeScript) = PrettyErgoTree.humanErgoTree(hexErgoTree)
+      F.pure(Right(ErgoTreeHuman(ergoTreeConstants, ergoTreeScript)))
```

Comment:
So from this code I can see that the API method is declared as infallible. But ErgoTreeConversionRequest is not validated and not guaranteed to be a valid ErgoTree (in encoded form). From user perspective it will be counter intuitive to handle errors as a special string in the response structure ("could not parse ergoTree" in this case). 
Wouldn't it be better to encode exception in the signature of `PrettyErgoTree.humanErgoTree`? And response with code 400 when an invalid ErgoTree is provided? Then users of this API can be sure that code 200 response contains successfully parsed ErgoTree.

Reply:
Very good point. I added bad request response for following 2 situations:
- string from request isn't in hex encoding
- hex ergo tree cannot be parsed for some reason